data/interp/davis/davis_data.py

```diff
@@ -2,23 +2,38 @@
 import os.path
 from data.interp.interp_data import InterpDataSet
 from utils.img import read_image
+from PIL import Image
+from io import BytesIO
 
 
 class DavisDataSet(InterpDataSet):
 
-    def __init__(self, directory, inbetween_locations, batch_size=1, maximum_shot_len=10):
+    def __init__(self, directory, inbetween_locations, batch_size=1):
         """
         See InterpDataSet.
         """
-        super().__init__(directory, inbetween_locations, batch_size=batch_size, maximum_shot_len=maximum_shot_len)
+        super().__init__(directory, inbetween_locations, batch_size=batch_size)
 
     def _process_image(self, filename):
         """
         Overriden.
-        TODO Consider cropping or downsizing the image first as they can be quite large.
         """
-        with open(filename, 'rb') as fp:
-            return fp.read()
+
+        # https://stackoverflow.com/questions/31826335/how-to-convert-pil-image-image-object-to-base64-string
+        buffered = BytesIO()
+        im = Image.open(filename)
+        width, height = im.size
+
+        # This just takes the whole image currently.
+        crop_width, crop_height = width, height
```

Comment:
This discards the rest of the image? Should it also take crops of the 4 corners if the crop size is small enough?

Reply:
This piece of cropping code here doesn't do anything. I just kept it for future reference, in case I do want to crop or resize images before saving to TFRecords.

Reply:
But if it were to do something, I think leaving out significant portions of the data would be sort of a data bug. I think it should either be fixed or be treated as dead code and deleted.